### DISCUSSION TITLE
Could somebody explain what a keyword pattern is? (With examples?)

### DISCUSSION THREAD

**Discussion by Feel-ix-343** [hrsh7th/nvim-cmp#1793] [Q&A] [answered]
I have not found real documentation on this.

**Comment by fitrh** [comment, marked as answer]
Try setting `nvim_lsp` source `keyword_pattern` to `[[\u]]`, then LSP completion will only be displayed if you type an uppercase character

```lua
cmp.setup({
  sources = {
    { 'nvim_lsp', keyword_pattern = [[\u]] },
  },
})
```

I don't fully understand this but from what I see, it is a vim regex to decide whether what we type triggers the completion or not
